### ISSUE TITLE
How to manage packages in R Markdown

### ISSUE THREAD

**Issue by aosavi** [yihui/rmarkdown-cookbook#219]
Trying to find out how to best manage the packages that are used in an R Markdown document. Such as when and where to install and load packages. Can't find anything other than [this](https://stackoverflow.com/questions/27941696/rmarkdown-why-do-i-have-to-reinstall-all-packages-everytime-i-knit). Might be something to describe in this book?

**Comment by cderv** [comment]
Managing packages for Rmarkdown is no more different that managing for R itself. I don't think there is anything specific to explain about Rmarkdown on this topic. 

The definitive guide already present R chunks and explain you can do anything R related in R code chunk: https://bookdown.org/yihui/rmarkdown/r-code.html

Explaining how to install and load package in R seems to be off topic for this book.

Do you have a specific usage in mind ? or a specific question ? 
What do mean by 
> best manage the packages that are used in an R Markdown document.

Thank you.

**Comment by emilyriederer** [comment]
(FYI @aosavi , this perhaps is not cookbook material, but in case it is useful to you I wanted to share something that I recently saw on [Twitter](https://twitter.com/MilesMcBain/status/1263272935197782016?s=20) :

![image](https://user-images.githubusercontent.com/19798371/83971779-bf8bcd00-a8a2-11ea-9565-45175273ce9f.png)

This won't tell you how to manage your packages, per say, but including such boilerplate code in an RMarkdown can create a "reproducibility receipt" that could help future users of your RMarkdown reconstruct your precise environment, if and when needed.)

**Comment by aosavi** [comment]
For some reason I suspected it to work differently from R. Not that I have found a completely satisfying way in R for collab and reproducibility :) but I agree that this book wouldn't be the place for that.

@emilyriederer , I like the idea of a reproducibility receipt, thanks! I'd also like to find a way to easily bring someones computer in the right 'state' for running the code, but maybe I should just use `renv` for that.
